src/Core/Checkout/Customer/Service/AddressHasher.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\Service;
+
+use Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\AddressHashStruct;
+use Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\CustomerAddressEntity;
+use Shopware\Core\Checkout\Customer\Event\AddressHashEvent;
+use Shopware\Core\Checkout\Order\Aggregate\OrderAddress\OrderAddressEntity;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Util\Hasher;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+#[Package('checkout')]
+class AddressHasher
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly EventDispatcherInterface $eventDispatcher,
+    ) {
+    }
+
+    public function isMatching(string $hash, CustomerAddressEntity|OrderAddressEntity $address): bool
+    {
+        return $hash === $this->generate($address);
+    }
```

Comment:
calling a public function in a decoratable class is highly discouraged. can't we do without this helper function? I mean, an `===` should be easily doable everywhere else.

Reply:
ik, was also my complaint doing `CartContextHasher`, but aligned with this here - I'm open for removing it

Reply:
don't remember this about that hasher over there, but yes, I would say let's remove it